core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -340,27 +354,60 @@ private String generatePublicKeyCert() {
   }
 
   // Schedules task to be executed once the provided futures are complete.
-  private <T> ListenableFuture<T> whenAllComplete(
-      Callable<T> task, ListenableFuture<?>... futures) {
+  private <T> ListenableFuture<T> whenAllSucceed(Callable<T> task, ListenableFuture<?>... futures) {
```

Comment:
Can you make this static and take the Executor as a parameter, since it's really meant to be a pure function?